soccer/src/soccer/strategy/agent/position/zoner.cpp

```diff
@@ -0,0 +1,126 @@
+#include "zoner.hpp"
+
+namespace strategy {
+
+Zoner::Zoner(int r_id) : Position(r_id, "Zoner") {}
+
+Zoner::Zoner(const Position& other) : Position{other} { position_name_ = "Zoner"; }
+
+std::optional<RobotIntent> Zoner::derived_get_task(RobotIntent intent) {
+    current_state_ = next_state();
+    // waller_id_ = get_waller_id();
+    return state_to_task(intent);
+}
+
+Zoner::State Zoner::next_state() {
+    rj_geometry::Point ball_pos = last_world_state_->ball.position;
+    rj_geometry::Segment block_path{ball_pos, field_dimensions_.our_goal_loc()};
+
+    double min_dist = std::numeric_limits<double>::infinity();
+    for (int i = 0; i < 6; i++) {
+        if (i == robot_id_) continue;
+        RobotState robot = last_world_state_->get_robot(true, i);
+        double dist = block_path.dist_to(robot.pose.position());
+        min_dist = std::min(min_dist, dist);
+    }
+
+    if (min_dist < (2 * kRobotRadius)) {
+        return ZONE;
+    } else {
```

Comment:
can replace with
```suggestion
   }
   return WALL; 
```
to fix GH warnings